src/dials/algorithms/indexing/indexer.py

```diff
@@ -613,6 +613,11 @@ def index(self):
 
                 if self._check_have_similar_crystal_models(experiments):
                     have_similar_crystal_models = True
+                    try:
+                        # Unset ids for the last experiment that is now deleted
+                        self._remove_id_from_reflections(len(experiments))
+                    except AttributeError:
```

Comment:
What is the scenario in which the above call would raise an `AttributeError`?

Reply:
I wondered if there is a circumstance in which `self.refined_reflections` does not exist because it's not actually defined until later. However I think the condition in this block is only true if there are already two crystals, in which case `self.refined_reflections` must have been created during the handling of the first one.

Maybe it would be better if `self.refined_reflections` was always created at the start of the function, even as an empty reflection table.

Reply:
In that case i think adding in the `except AttributeError` is misleading, and we should either define self.refined_reflections at the start or add a comment to explain that this is only called once `self.refined_reflections` exists.